src/platforms/web/runtime/modules/style.js

```diff
@@ -1,18 +1,19 @@
 /* @flow */
 
 import { getStyle, normalizeStyleBinding } from 'web/util/style'
-import { cached, camelize, extend, isDef, isUndef } from 'shared/util'
+import { cached, camelize, extend, isDef, isUndef, hyphenate } from 'shared/util'
 
 const cssVarRE = /^--/
 const importantRE = /\s*!important$/
 const setProp = (el, name, val) => {
+  const hyphenatedName = hyphenate(name)
   /* istanbul ignore if */
-  if (cssVarRE.test(name)) {
-    el.style.setProperty(name, val)
+  if (cssVarRE.test(hyphenatedName)) {
+    el.style.setProperty(hyphenatedName, val)
   } else if (importantRE.test(val)) {
-    el.style.setProperty(name, val.replace(importantRE, ''), 'important')
+    el.style.setProperty(hyphenatedName, val.replace(importantRE, ''), 'important')
```

Comment:
We only need hyphenated name here. Other modification can be discarded.

```suggestion
    el.style.setProperty(hyphenate(name), val.replace(importantRE, ''), 'important')
```

Reply:
Thanks for the review, this was also my first thought but considering that setProperty requires a hyphenated parameter I don't see a problem of letting this change happen in all invocations. Also, in line 16 the normalize function will convert it to camelCase every time so I don't see a problem there too. If we ultimately want to keep the scope of the change smaller as possible I agree with your proposal but if not I think is better for consistency to not have different versions of the same information spread over the function.

Reply:
It's mostly performance. If we can avoid the hyphenate call in some branches then we should.